lib/python/treadmill_aws/hostmanager.py

```diff
@@ -46,7 +46,13 @@ def create_host(ec2_conn, ipa_client, image_id, count, domain,
     hosts = []
     for _ in range(count):
         host_ctx = instance_vars.copy()
-        host_ctx['hostname'] = generate_hostname(domain=domain, image=image_id)
+
```

Comment:
well, if anything, the check for count == len should be here, not in CLI.

this function does way too much imo. need to split.